src/pudl_archiver/archivers/eia923.py

```diff
@@ -22,14 +21,13 @@ async def get_resources(self) -> ArchiveAwaitable:
         link_pattern = re.compile(r"f((923)|(906920))_(\d{4})\.zip")
 
         for link in await self.get_hyperlinks(BASE_URL, link_pattern):
-            yield self.get_year_resource(link, link_pattern.search(link))
+            year = int(link_pattern.search(link).group(4))
```

Comment:
I was trying to think of a way to have the base class handle checking the year without having to call `valid_year` from every subclass, but I don't see any particularly straightforward ways to do that, so this seems fine

Reply:
Yeah, I had been hoping for this too but because different datasets handle years differently, and some have XBRL/DBF and some don't, it just seemed easier to put it in each subclass. If we figure something out later it's always easier to add abstraction than remove it.